### PR TITLE
HTTPClient

### PR DESCRIPTION
**Pull request by VasylDobrovolskyi**
Basic implementation of HTTPClient.

### CODE REVIEW

File: src/Client/HTTPClient.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        std::ostream& o = session.sendRequest(request);
+        o << net_request.json_playload.toJson().constData();
+    }
+    else if (net_request.content_type == Net::CONTENT_TYPE_PLAIN_TEXT)
+        ;
+    else if (net_request.content_type == Net::CONTENT_TYPE_EMPTY)
+        ;
+    else
+    {
+        throw ("Unsupported content type");
```
Comment (Denys-Bykov-0) [inline src/Client/HTTPClient.cpp:41]: Replace it with exception class inherited from our base class "Exception"

File: src/Client/HTTPClient.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    Poco::Net::HTTPClientSession session(uri.getHost(), uri.getPort());
+    std::string path(uri.getPathAndQuery());
+    Poco::Net::HTTPRequest request(net_request.method, path, Poco::Net::HTTPMessage::HTTP_1_1);
+    SendRequest(request, session, net_request);
+    Poco::Net::HTTPResponse response;
+    std::istream &received_stream = session.receiveResponse(response);
+    return FormResponse(response, received_stream);
+}
+
+void HTTPClient::AddCredentials(Poco::Net::HTTPRequest &request,
```
Comment (Denys-Bykov-0) [inline src/Client/HTTPClient.cpp:86]: net_request parameter is redundunt, please remove it.

File: src/Client/HTTPClient.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    }
+    else
+    {
+        throw Exception("Unsupported content type");
+    }
+}
+
+Net::Response HTTPClient::FormResponse(
+    const Poco::Net::HTTPResponse& response,
+    std::istream &received_stream)
```
Comment (iodream) [inline src/Client/HTTPClient.cpp:52]: as per code convention, prefer & adjacent to type
```suggestion
    std::istream& received_stream)
```

File: src/Client/HTTPClient.cpp [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+Net::Response HTTPClient::Request(const Net::Request& net_request)
+{
+    Poco::URI uri(net_request.uri);
+    Poco::Net::HTTPClientSession session(uri.getHost(), uri.getPort());
+    std::string path(uri.getPathAndQuery());
+    Poco::Net::HTTPRequest request(net_request.method, path, Poco::Net::HTTPMessage::HTTP_1_1);
+    SendRequest(request, session, net_request);
+    Poco::Net::HTTPResponse response;
+    std::istream &received_stream = session.receiveResponse(response);
```
Comment (iodream) [inline src/Client/HTTPClient.cpp:87]:
```suggestion
    std::istream& received_stream = session.receiveResponse(response);
```

File: src/Client/HTTPClient.cpp [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    return FormResponse(response, received_stream);
+}
+
+void HTTPClient::AddCredentials(Poco::Net::HTTPRequest &request)
```
Comment (iodream) [inline src/Client/HTTPClient.cpp:91]:
```suggestion
void HTTPClient::AddCredentials(Poco::Net::HTTPRequest& request)
```

**Review by iodream — APPROVED** [review]
seems to be ok

**Review by Denys-Bykov-0 — APPROVED** [review]

**Pull request merged**